Add endpoint for set attestation policy (#81)
confidential-containers/kbs: docs/kbs_attestation_protocol.md

2. `/kbs/v0/attest` only accepts `POST` requests whose body is a [KBS Attestation](#attestation)
   JSON payload and the header contains a `Cookie` set to the value received in
   step 1.i. This is how the attester replies to attestation challenge received
   in step 1.ii. If the attestation was successful, the KBS replies to that request with an 

>    [Attestation Results Token](#attestation-results-token).

Reviewer: If I understand correctly, this is not related to the attestation policy setting. If that's so, could we have a separate PR for this fix, please?

Author: OK, I'll do this work.

party service API, then the symmetric key used to encrypt the output payload can
be wrapped with the provided `tee-pubkey`.

### Set Attestation Policy

> User of KBS can set attestation verification policy with posting following endpoint:

Reviewer: ```suggestion
User of KBS can set an attestation verification policy through the following endpoint:
```

```
/kbs/v0/attestation-policy
```

The payload of the POST request should like:

```json
{
    "type": "rego",
    "policy_id": "default",
    "policy": <base64encoded policy>
}
```

Where `type` specified the policy format, `policy_id` specified the ID of the policy
and `policy` is the base64 encoded policy content.

POST to this endpoint needs user identity authentication with carrying JWT (JSON Web Token)

> signed by the user's private key in the Request.

Reviewer: So the attestation policy is global to the whole KBS? Are we expecting the policy to describe e.g. different policies for different resources?

Author: The policy here is the policy for verifying evidence, and in the RATS model, it should be the policy of the Verifier. And the policy you mentioned should be a policy of the rely party, which distributes different confidential resources based on the attestation results.

I think it's best for us to separate these two policies, although we haven't implemented the latter yet.

Reviewer: Ok, that makes sense to me. Thanks for the clarification.
